src/components/ec-metroline-item/ec-metroline-item.vue

```diff
@@ -103,69 +103,62 @@
   </div>
 </template>
 
-<script>
+<script setup>
+import {
+  computed, inject, onBeforeUnmount, onMounted, toRefs,
+} from 'vue';
+
 import * as MetrolineItemStatus from '../../enums/metroline-item-status';
 import EcIcon from '../ec-icon';
 
-export default {
-  name: 'EcMetrolineItem',
-  components: { EcIcon },
-  inject: ['metroline'],
-  props: {
-    id: {
-      type: Number,
-      required: true,
-    },
-    badgeText: {
-      type: String,
-    },
-  },
-  computed: {
-    isReadOnly() {
-      return this.metroline.isCompleted;
-    },
-    isLast() {
-      return this.id === this.metroline.lastItemId;
-    },
-    status() {
-      if (this.id < this.metroline.activeItemId || this.metroline.isCompleted) {
-        return MetrolineItemStatus.COMPLETED;
-      }
-
-      if (this.id === this.metroline.activeItemId) {
-        return MetrolineItemStatus.ACTIVE;
-      }
-
-      return MetrolineItemStatus.NEXT;
-    },
-    isNext() {
-      return this.status === MetrolineItemStatus.NEXT;
-    },
-    isActive() {
-      return this.status === MetrolineItemStatus.ACTIVE;
-    },
-    isCompleted() {
-      return this.status === MetrolineItemStatus.COMPLETED;
-    },
-  },
-  created() {
-    this.metroline.register(this.id);
-  },
-  beforeUnmount() {
-    this.metroline.unregister(this.id);
+const metroline = inject('metroline');
```

Comment:
If we use provide/inject pattern it should be only situation where components are tightly coupled. If we decided that `ec-metroline-item` and `ec-metroline` component are such tightly coupled, then I believe they should be placed in one directory named `ec-metroline`, and `ec-metroline-item` should be placed in subdirectory `elements` as it could be used only inside `ec-metroline`